File: LeetCode Solutions/Decision Tree/Calculate Entropy.py
class Solution:
    def calculateEntropy(self, input: List[int]) -> float:
        n = len(input)
        nums = set(input)
        entropy = 0.
        for num in nums:
            p_i = input.count(num)/n
            entropy -= p_i * math.log2(p_i)
        return entropy

File: LeetCode Solutions/Decision Tree/Calculate Entropy.cpp
class Solution {
public:
    double calculateEntropy(vector<int>& input) {
        unordered_map<int, int> umap;
        for(auto tmp: input) {
            umap[tmp]++;
        }
        double size = input.size(), result = 0, px;
        for(auto tmp: umap) {
            px = (tmp.second) / size;
            result -= px*log2(px);
        }
        return result;
    }
};